code/modules/admin/admin_memo.dm

```diff
@@ -8,8 +8,17 @@
 	if(!ENABLE_MEMOS)		return
 	if(!check_rights(0))	return
 	switch(task)
-		if("write")		admin_memo_write()
-		if("show")		admin_memo_show()
+		if("write")
+			switch(input("Is this a temporary note? If so, it will auto-erase at the end of the current round.","Memo")in list("Yes","No","Cancel"))
+				if("Yes")
+					var/msg = input("","Memo") as message
+					Shift_Transition_Memo += "	<CENTER><I>([time2text(world.realtime,"(DDD) DD MMM hh:mm")])<BR><B>[usr.ckey]([usr.name]):</B></CENTER><BR> [msg] "
+					message_admins("[key] set a temporary memo.")
+				if("No")	admin_memo_write()
+		if("show")
+			switch(input("Show which memo?","Memo")in list("Admin Memo","Temp Memo","Cancel"))
+				if("Admin Memo")	admin_memo_show()
```

Comment:
Looks much better. These huge spaces in the code are weird, though.

I'll be able to test it tomorrow.